dali/dfu/dfuwu.cpp

```diff
@@ -3287,7 +3287,7 @@ class CDFUWorkUnitFactory : implements IDFUWorkUnitFactory, implements ISDSSubsc
     {
         StringBuffer path("*");
         if (owner && *owner)
-            path.append("[@submitID=\"").append(owner).append("\"]");
+            path.append("[@submitID=?~\"").append(owner).append("\"]");
```

Comment:
Is the '~' to specify a wild match?  If so, that MIGHT not be needed anymore (ptree might always support wild card characters?).  but I guess it doesn't hurt.

Reply:
As I said doesn't hurt and makes the wildcard more explicit.

Reply:
the implication is that wildcard is desired here? (it would not have done so before with a regular equality qualifier)

I would not expect getWorkUnitsByOwner("usera") to also return dfu workunits owned by "notusera" ..

I do notice it's already this way for ecl workunits in it's implementation of getWorkUnitsByOwner, but should it be?

Reply:
Also, using anything but a regular equality is lot more inefficient, because jptree optimizes lookups of sibling via qualifiers on attributes with straight equalities (by auto building HT's on the fly), so using "=?" (or =?~") will result in a linear scan of all dfu workunits instead.

The normal approach to case-insensitivity employed elsewhere is to store it lower case it when set, and looked up.
Perhaps the original cased submitID is needed here(?), but we could add an internal attribute for this purpose instead and thus avoid "=?".
Either way, that won't help with pre-existing workunits that don't have the lower-case @submitID (but could be 'fixed' automatically via other changes).

@wangkx - first question is - does submitID need to preserve case when it is set?

Reply:
@jakesmith I have 3 concerns for not preserving case. 1. confuse a user ('who changed my username?' or 'the submitID was not in lower case on previous ECLWatch page'). 2. how to handle old WUs which contain case sensitive submitID? 3. break some existing logic (where? I do not know).

Reply:
@wangkx - this is quite a significant performance issue.

+ @submitID doesn't need to change.
+ A new internal attribute can be added (lower-cased) explicitly for the purpose of optimized caseless searching.
+ Old WUs can be scanned for the new missing attribute at startup time (it would be best to do in Dali itself), and the new attribute added.
+ Code for searching attributed can use regular hard equality matching, which will be an order of magnitude faster than the linear search that =? would involve.

Reply:
Not sure why this one was marked as resolved.
There are 2 relates issues here:

1) the wildcard question - why should getWorkUnitsByOwner("usera") return results for "notusera" ?
2) the efficiency issues caused by not performing a standard equality match